### ISSUE TITLE
change only icon color

### ISSUE THREAD

**Issue by zhengpd** [nvim-tree/nvim-web-devicons#444]
It would be nice if the icon can change only color with something like `require"nvim-web-devicons".set_icon_color("rb", "#00ff00")`

**Comment by alex-courtis** [comment]
Yes, that would be useful. It seems that you would need to get first, change then set.

**Comment by gegoune** [comment]
This should be achievable today with what Alex is suggesting. Something like:

```lua
require'nvim-web-devicons'.get_icons()['a']
```
or
```lua
require'nvim-web-devicons'.get_icons_by_filename()['webpack']
```
can be used to get icon's table which can be modified however you like and returned with https://github.com/nvim-tree/nvim-web-devicons?tab=readme-ov-file#set-an-icon.

I haven't tested it. @zhengpd Would that be sufficient for your need?

@alex-courtis Or do we want to introduce API to retrieve single icon only?

**Comment by alex-courtis** [comment]
> [@alex-courtis](https://github.com/alex-courtis) Or do we want to introduce API to retrieve single icon only?

That sounds good; it's probably what should have been done instead of returning icon/higroup.

It'll need a different name; `get_icon_full`?

**Comment by gegoune** [comment]
We could unify `get_icon_color()` and `get_icon()` into new `icon()` returning table with all keys and deprecate old methods.
